### DISCUSSION TITLE
How often does the access token refresh in H2O Wave when using OIDC?

### DISCUSSION THREAD

**Discussion by nipunchamikara** [h2oai/wave#2270] [Q&A] [answered]
Does Wave refresh the access token every time the user performs an action or when the access token has expired?

**Comment by mturoci** [comment, marked as answer]
Hi @nipunchamikara.

Wave refreshes access token on every user action if necessary. One can also [refresh the token explicitly](https://wave.h2o.ai/docs/security/#explicit-token-refresh) if needed.
